components/movementsensor/adxl345/adxl345.go

```diff
@@ -0,0 +1,298 @@
+	rawConfig config.Component,
+	logger golog.Logger,
+) (movementsensor.MovementSensor, error) {
+	cfg := rawConfig.ConvertedAttributes.(*AttrConfig)
+	b, err := board.FromDependencies(deps, cfg.BoardName)
+	if err != nil {
+		return nil, err
+	}
+	localB, ok := b.(board.LocalBoard)
+	if !ok {
+		return nil, errors.Errorf("board %s is not local", cfg.BoardName)
+	}
+	bus, ok := localB.I2CByName(cfg.I2cBus)
+	if !ok {
+		return nil, errors.Errorf("can't find I2C bus '%s' for ADXL345 sensor", cfg.I2cBus)
+	}
+
+	var address byte
+	if cfg.UseAlternateI2CAddress {
+		address = 0x1D
+	} else {
+		address = 0x53
+	}
+
+	backgroundContext, cancelFunc := context.WithCancel(ctx)
+
+	sensor := &adxl345{
+		bus:               bus,
+		i2cAddress:        address,
+		logger:            logger,
+		backgroundContext: backgroundContext,
+		cancelFunc:        cancelFunc,
+	}
+
+	// To check that we're able to talk to the chip, we should be able to read register 0 and get
+	// back the device ID (0xE5).
+	deviceID, err := sensor.readByte(ctx, 0)
+	if err != nil {
+		return nil, errors.Errorf("can't read from I2C address %d on bus %s of board %s: '%s'",
+			address, cfg.I2cBus, cfg.BoardName, err.Error())
+	}
+	if deviceID != 0xE5 {
+		return nil, errors.Errorf("unexpected I2C device instead of ADXL345 at address %d: deviceID '%d'",
+			address, deviceID)
+	}
+
+	// The chip starts out in standby mode. Set it to measurement mode so we can get data from it.
+	// To do this, we set the Power Control register (0x2D) to turn on the 8's bit.
+	err = sensor.writeByte(ctx, 0x2D, 0x08)
+	if err != nil {
+		return nil, errors.Errorf("unable to put ADXL345 into measurement mode: '%s'", err.Error())
+	}
+
+	// Now, turn on the background goroutine that constantly reads from the chip and stores data in
+	// the object we created.
+	sensor.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(sensor.pollData)
+
+	return sensor, nil
+}
+
+func (adxl *adxl345) readByte(ctx context.Context, register byte) (byte, error) {
+	result, err := adxl.readBlock(ctx, register, 1)
+	if err != nil {
+		return 0, err
+	}
+	return result[0], err
+}
+
+func (adxl *adxl345) readBlock(ctx context.Context, register byte, length uint8) ([]byte, error) {
+	handle, err := adxl.bus.OpenHandle(adxl.i2cAddress)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := handle.Close()
+		if err != nil {
+			adxl.logger.Error(err)
+		}
+	}()
+
+	results, err := handle.ReadBlockData(ctx, register, length)
+	return results, err
+}
+
+func (adxl *adxl345) writeByte(ctx context.Context, register, value byte) error {
+	handle, err := adxl.bus.OpenHandle(adxl.i2cAddress)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := handle.Close()
+		if err != nil {
+			adxl.logger.Error(err)
+		}
+	}()
+
+	return handle.WriteByteData(ctx, register, value)
+}
+
+// A helper function: takes 2 bytes and reinterprets them as a little-endian signed integer.
+func toSignedValue(data []byte) int {
```

Comment:
can you use one of the utils/math.go function instead? if it doesn't exists might make sense to add it